### ISSUE TITLE
Creating a group chat with yourself

### ISSUE THREAD

**Issue by avdoseferovic** [Expensify/Expensify.cash#4011]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. login with valid credentials
2. click the plus sign
3. click new chat
4. insert a phone number you own that's not linked to that account
5. open the conversation
6. send a message
7. go to profile settings
8. link that phone number to your account
9. the initial conversation stays in the left sidebar

Now you can chat with yourself, organize meet/zooms.
Main issue is that when you request money from 'number' conversation you get stuck in a infintite loading screen.

## Expected Result:
Apart from allowing to send messages to unregistered numbers (which I see ways to abuse, lets say you registerd on the app and start sending random messages to random numbers, even with a limit this would be easily abusable as you actually get a sms message without being registered), you shouldn't be able to create a room with yourself.

## Actual Result:
Conversation of your own number is stuck and you can actually text with yourself,

## Workaround:
Remove the conversation if the texted phone number is linked to the same profile.

## Platform:
Where is this issue occurring?

 Web ✔️
 iOS 
 Android✔️
 Desktop App
 Mobile Web✔️

**Version Number:**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @lschurr (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by avdoseferovic** [comment]
I'd love to work on this if this is an unexpected behaviour.

**Comment by rdjuric** [comment]
I really like being able to have a chat with myself on Slack/WhatsApp. Useful for notes and testing stuff.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @alex-mechler (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by lschurr** [comment]
I'm not super well-versed on the roadmap and whether this is something we're currently working on. Throwing the eng label for more insight.

**Comment by alex-mechler** [comment]
Thanks for reporting this! I talked about this with @trjExpensify and we decided that while this is not the most intuitive result, its one that does not negatively impact the users experience. If a user wants to go out of their way to create a chat with themselves this way, we don't need to block it. We do need to block someone sending themselves money, which we are handling [in this issue](https://github.com/Expensify/Expensify.cash/issues/3572).
